Modules/_json.c

```diff
@@ -413,6 +413,7 @@ scanstring_unicode(PyObject *pystr, Py_ssize_t end, int strict, Py_ssize_t *next
         if (c == '"') {
             // Fast path for simple case.
             if (_PyUnicodeWriter_IsEmpty(writer)) {
+                PyUnicodeWriter_Discard(writer);
```

Comment:
Shouldn't you do this *after* the if statement (which might trigger goto jump, that also calls PyUnicodeWriter_Discard())?

Reply:
Right, goto bail calls Discard() again. Either set writer to NULL or move Discard() below.

Reply:
Fixed